File: C#/Laborator3/service/Service.cs
using Laborator4.domain;
using Laborator4.repository;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Laborator4.service
{
    public class Service
    {

        AngajatDBRepository angajatRepo;
        ZboruriDBRepository zborRepo;
        BiletDBRepository biletRepo;

        public Service(AngajatDBRepository angajatRepo,
                            ZboruriDBRepository zborRepo,
                            BiletDBRepository biletRepo)
        {
            this.angajatRepo = angajatRepo;
            this.zborRepo = zborRepo;
            this.biletRepo = biletRepo;
        }
        public Angajat findOne(String username)
        {
            return angajatRepo.FindOne(username);
        }

        public IEnumerable<Zbor> findAllZboruri()
        {
            return zborRepo.FindAll();
        }

        public IEnumerable<Bilet> findAllBilete()
        {
            return biletRepo.FindAll();
        }

        public IEnumerable<Angajat> findAllAngajati()
        {
            return angajatRepo.FindAll();
        }

        public void addBilet(Bilet bilet)
        {
            biletRepo.Save(bilet);
            Zbor zbor = zborRepo.findOne(bilet.zborId);
            zbor.nrLocuriDisponibile=zbor.nrLocuriDisponibile - bilet.nrLocuri;
            zborRepo.Update(zbor);
        }



        public Angajat getAngajat(String username, String parola)
        {
            Angajat a = null;
            IEnumerable<Angajat> angajati = findAllAngajati();
            angajati.ToList().ForEach(
                x =>
                {
                    if (x.id.Equals(username) && x.parola.Equals(parola))
                        a = x;
                });
            return a;
        }

        public IEnumerable<Zbor> FilterByDestinatie(String aeroport, DateTime data)
        {
            
            return zborRepo.FilterByDestinatie(aeroport, data);
        }

    }
}

File: Java/src/main/java/service/Service.java
package service;

import domain.Angajat;
import domain.Bilet;
import domain.Zbor;
import repository.AngajatDBRepository;
import repository.BiletDBRepository;
import repository.ZborDBRepository;

import java.time.LocalDate;
import java.util.List;

public class Service {

    private AngajatDBRepository angajatDBRepository;
    private ZborDBRepository zborDBRepository;
    private BiletDBRepository biletDBRepository;

    public Service(AngajatDBRepository angajatDBRepository, ZborDBRepository zborDBRepository, BiletDBRepository biletDBRepository) {
        this.angajatDBRepository = angajatDBRepository;
        this.zborDBRepository = zborDBRepository;
        this.biletDBRepository = biletDBRepository;
    }

    public Angajat findOne(String username){
        return angajatDBRepository.findOne(username);
    }

    public Iterable<Angajat> findAll() {
        return angajatDBRepository.findAll();
    }

    public Angajat getAngajat(String username, String parola) {
        for (Angajat a:findAll())
            if(a.getId().equals(username) && a.getParola().equals(parola))
                return a;
        return null;
    }

    public Iterable<Zbor> getAllZboruri() {
        return zborDBRepository.findAll();
    }

    public Iterable<Bilet> getAllBilete() {
        return biletDBRepository.findAll();
    }

    public void addBilet(Bilet bilet){
        biletDBRepository.save(bilet);
        Zbor zbor=zborDBRepository.findOne(bilet.getZborId());
        zbor.setNrLocuriDisponibile(zbor.getNrLocuriDisponibile()-bilet.getNrLocuri());
        zborDBRepository.update(zbor);
    }

    public List<Zbor> filterByAeroport(String aeroport, LocalDate data){
        return zborDBRepository.filterByAeroport(aeroport, data);
    }
}
